DllImportGenerator/DllImportGenerator/StubCodeGenerator.cs

```diff
@@ -138,31 +144,44 @@ public override (string managed, string native) GetIdentifiers(TypePositionInfo
             // Stub return is not the same as invoke return
             if (!stubReturnsVoid && !retMarshaller.TypeInfo.IsManagedReturnPosition)
             {
+                // Stub return should be the last parameter for the invoke
                 Debug.Assert(paramMarshallers.Any() && paramMarshallers.Last().TypeInfo.IsManagedReturnPosition);
 
-                // Declare variable for stub return value
-                TypePositionInfo info = paramMarshallers.Last().TypeInfo;
-                statements.Add(LocalDeclarationStatement(
-                    VariableDeclaration(
-                        info.ManagedType.AsTypeSyntax(),
-                        SingletonSeparatedList(
-                            VariableDeclarator(this.GetIdentifiers(info).managed)))));
+                (TypePositionInfo stubRetTypeInfo, IMarshallingGenerator stubRetGenerator) = paramMarshallers.Last();
```

Comment:
Can we assert some invariant here? Like we know we are in a `PreserveSig` scenario?

Reply:
Mm, the only thing I had was the assert above about the last parameter being the managed return.

We could give the `StubCodeGenerator` the `GeneratedDllImportData` so it knows about `PreserveSig` and can explicitly assert on that. It will need to know about `SetLastError` when we support that, so giving it all the data seems fine.